### PR TITLE
Add documentation for v9 SQL creation and connection

### LINKED ISSUE #34: V9 Migration SQL Creation and Connection Documentation
As part of [teammate's v9 ](https://github.com/TEAMMATES/teammates/issues/12048), documentation to create SQL instances and connect them on staging is necessary.

### PR DESCRIPTION
**Pull request by FergusMok**
Closes #34.

1. Separated VPC connectors into it's own section
2. Added section for SQL creation, connection, debugging and minimum-privileged user.
3. Updated some missing parts and removed deprecated APIs.

### CODE REVIEW

File: platform-guide.md [hunk 1]
```diff
@@ -29,7 +31,8 @@ Note: This document does not have preference over either GAE standard or flexibl
 1. Enable the following APIs in your project:
    - [Cloud Tasks API](https://console.cloud.google.com/apis/library/cloudtasks.googleapis.com)
    - [Cloud Scheduler API](https://console.cloud.google.com/apis/library/cloudscheduler.googleapis.com)
-   - [Cloud Debugger API](https://console.cloud.google.com/apis/library/clouddebugger.googleapis.com)
```
Comment (FergusMok) [inline platform-guide.md:32]: Cloud debugger was shut down last year
https://cloud.google.com/stackdriver/docs/deprecations/debugger-deprecation

File: platform-guide.md [hunk 2]
```diff
@@ -56,6 +63,9 @@ Note: This document does not have preference over either GAE standard or flexibl
    * Run the following command:
 
      ```sh
+     # Creates the application on GCP
+     gcloud app create
```
Comment (FergusMok) [inline platform-guide.md:67]: Need to create first, otherwise `appengineDeployAll` will fail

**Comment by FergusMok** [comment]
Hi @wkurniawan07, I've added the portion on creating the user with minimum privileges. 

For the remaining portion on data migrations (creation, edit) and changelogs using Liquibase, it'll be followed up in a future PR by @NicolasCwy

**Review by cedricongjh — APPROVED** [review]
LGTM!

**Review by ziqing26 — APPROVED** [review]
Thanks for putting up the docs!

**Pull request merged**